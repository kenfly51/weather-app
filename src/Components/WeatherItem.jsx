import React from "react";
import PropTypes from "prop-types";
import Card from "react-bootstrap/Card";
import { getStateImage } from "../helpers/paths";
import { getDayName, formatTemp } from "../helpers/helpers";

const WeatherItem = ({ item }) => {
  const { date, min, max, state, stateAbbr } = item;

  const dayOfWeek = getDayName(date);
  return (
    <Card className="weather-item">
      <Card.Img
        className="state-image"
        variant="top"
        src={getStateImage(stateAbbr)}
      />
      <Card.Body>
        <Card.Title>{dayOfWeek}</Card.Title>
        <Card.Subtitle>{state}</Card.Subtitle>
        <div className="temp-details">
          <div>{`min: ${formatTemp(min)}`}</div>
          <div>{`max: ${formatTemp(max)}`}</div>
        </div>
      </Card.Body>
    </Card>
  );
};

WeatherItem.propTypes = {
  item: PropTypes.shape({
    date: PropTypes.instanceOf(Date).isRequired,
    min: PropTypes.number.isRequired,
    max: PropTypes.number.isRequired,
    temp: PropTypes.number.isRequired,
    state: PropTypes.string.isRequired,
    stateAbbr: PropTypes.string.isRequired,
  }).isRequired,
};

export default WeatherItem;
