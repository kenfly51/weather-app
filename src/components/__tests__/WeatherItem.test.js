import { render } from "@testing-library/react";
import WeatherItem from "../WeatherItem";

describe("WeatherItem", () => {
  it("should render WeatherItem correctly", () => {
    const { container } = render(
      <WeatherItem
        item={{
          date: new Date("2021-01-26"),
          min: 11.2,
          max: 11.9,
          state: "state",
          stateAbbr: "ste",
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
