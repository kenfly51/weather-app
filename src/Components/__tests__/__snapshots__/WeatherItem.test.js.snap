// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItem should render WeatherItem correctly 1`] = `
<div>
  <div
    class="weather-item card"
  >
    <img
      class="card-img-top state-image"
      src="https://www.metaweather.com/static/img/weather/ste.svg"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Tuesday
      </div>
      <div
        class="card-subtitle h6"
      >
        state
      </div>
      <div
        class="temp-details"
      >
        <div>
          min: 11
        </div>
        <div>
          max: 11
        </div>
      </div>
    </div>
  </div>
</div>
`;
